### PR TITLE
[Feat]: Provide Sequence as a Backend for APIs

### PR DESCRIPTION
**Pull request by BLasan**
Provide a separate Endpoint option for REST APIs to define a sequence as its backend. This addresses the following use-cases.

1. Add new Sequence for Production and Sandbox key types.
2. Delete a sequence
3. Download a sequence

UI will now look as follows for the REST APIs.
Screenshot 2024-09-25 at 01 04 55
Screenshot 2024-09-25 at 01 06 41

### Related Issues
- https://github.com/wso2/api-manager/issues/3007

### CODE REVIEW

File: portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx [hunk 1]
```diff
@@ -0,0 +1,621 @@
+/* eslint-disable */
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (RakhithaRR) [inline portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx:3]: Let's change this to the new license text

File: portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+                        setIsValidSequenceBackend(false);
+                    }
+                }
+            });
+        }
+    }
+
+
+    /**
+     * Method to upload the certificate content by calling the rest api.
```
Comment (RakhithaRR) [inline portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx:261]: Shall we fix add the correct method comments

File: portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx [hunk 3]
```diff
@@ -0,0 +1,621 @@
...
+    const closeCustomBackendUpload = () => {
+        setUploadCustomBackendOpen({ open: false, keyType: '' });
+        setCustomBackend({ name: '', content: '' });
+    };
+
+    useEffect(() => {
+        setIsValidSequenceBackend(false);
+        restAPI.getSequenceBackends(api.id)
+            .then((result) => {
+                console.log(result);
```
Comment (Krishanx92) [inline portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx:190]: remove console log

File: portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx [hunk 4]
```diff
@@ -0,0 +1,621 @@
...
+     *
+     * @param {any} event The button click event.
+     * @param {string} certAlias  The alias of the certificate which information is required.
+     * */
+    const showSequenceBackendDeleteDialog = async (event, keyType, name) => {
+        setSequenceBackendToDelete({ open: true, keyType: keyType, name: name });
+    };
+
+    const downloadCustomBackend = (keyType) => {
+        console.log("Downloading");
```
Comment (Krishanx92) [inline portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx:215]: remove and check all other places

File: portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx [hunk 5]
```diff
@@ -0,0 +1,621 @@
...
+     * */
+    const saveCustomBackend = () => {
+        setSaving(true);
+        if (uploadCustomBackendOpen.keyType === API_SECURITY_KEY_TYPE_SANDBOX) {
+            sandBoxBackendList.push({"sequenceName": customBackend.name, "content": customBackend.content});
+        } else {
+            productionBackendList.push({"sequenceName": customBackend.name, "content": customBackend.content});
+        }
+        restAPI.uploadCustomBackend(customBackend.content, uploadCustomBackendOpen.keyType, api.id).then((resp) => {
+            console.log('Custom backend uploaded successfully' + resp);
```
Comment (Krishanx92) [inline portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/CustomBackend.jsx:271]: we need messge box not console log

File: portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx [hunk 6]
```diff
@@ -298,6 +314,9 @@ function EndpointOverview(props) {
                 { key: 'awslambda', value: 'AWS Lambda' },
             ];
         }
+        if(apiObject.gatewayType !== 'wso2/apk' && type === 'HTTP' ) {
```
Comment (Krishanx92) [inline portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx:317]: formatting

**Comment by Krishanx92** [comment]
Can you attached new UI design

**Review by Krishanx92 — CHANGES_REQUESTED** [review]

**Comment by BLasan** [comment]
> Can you attached new UI design

Attached in the description. https://github.com/wso2/apim-apps/pull/743#issue-2527489091

**Review by Krishanx92 — APPROVED** [review]

**Review by Krishanx92 — APPROVED** [review]

**Review by RakhithaRR — APPROVED** [review]

**Pull request merged**